### DISCUSSION TITLE
Container Registry Package Description

### DISCUSSION THREAD

**Discussion by mhoran** [community/community#26565] [Packages]
<p>The GitHub container registry is great! I’ve managed to publish my containers with the provided GitHub workflow. However, when I look at my container’s package page, the “package description” is empty – despite my project’s README.md. I haven’t been able to figure out a way to provide a description. I tried setting some metadata in my Dockerfile, but that didn’t work. I can’t find any option to edit the description from package settings. So how am I supposed to edit my container’s description?</p>
<p>Thanks!</p>

**Comment by clarkbw** [comment, marked as answer]
<p>That’s great to hear! :tada:</p>
<p>We’re working on better representing your README at the container page view. For now you can set a smaller description via a Docker label which will show up on the container page.</p>
<p>Add the following label to your <code>Dockerfile</code>:</p>
<pre><code class="lang-auto">LABEL org.opencontainers.image.description="A short text-only description that will appear on the containers page."
</code></pre>
<p>Or you can also add this label at build time:</p>
<pre><code class="lang-shell">docker build --label "org.opencontainers.image.description=A short text-only description that will appear on the containers page."
</code></pre>

**Reply by alimd** [reply]
Not work at all!

**Reply by kevincam3** [reply]
This doesn't work anymore.

**Comment by mhoran** [comment]
<p>That did it, thanks!</p>

**Comment by airtower-luna** [comment]
<p>Is there documentation anywhere for which labels are recognized for which purposes? I’ve seen you recommend them in a bunch of posts here, but I can’t find any of them on <a href="http://docs.github.com">docs.github.com</a>. The <a href="https://github.com/opencontainers/image-spec/blob/79b036d80240ae530a8de15e1d21c7ab9292c693/annotations.md">Open Containers spec</a> describes a bunch of annotations, but naturally doesn’t tell me which are used by GHCR and how. It would be very helpful to have a summary. :slightly_smiling_face:</p>

**Comment by clarkbw** [comment]
<p>New docs are up for the docker labels required to connect to a repo. We haven’t changed the label, it’s still the <code>org.opencontainers.image.source</code> label but now we have a page describing how this works <a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/connecting-a-repository-to-a-container-image">https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/connecting-a-repository-to-a-container-image</a></p>

**Comment by gcstr** [comment]
<p>Hi <a class="mention" href="/clarkbw">@clarkbw</a>,</p>
<p>In your example you mentioned</p>
<pre><code class="lang-auto">LABEL org.opencontainers.image.description="A short text-only description that will appear on the containers page." 
</code></pre>
<p>When you say</p>
<blockquote>
<p>short text-only description</p>
</blockquote>
<p>How short you mean? What is the size limit of the string?</p>

**Comment by clarkbw** [comment]
<p>Most labels (description included) are limited to <code>512</code> characters except for licenses which is limited to <code>256</code></p>

**Comment by clarkbw** [comment]
<p>I opened an issue with the docs repo to get this created and it has the details on the other LABELs which can be used. <a href="https://github.com/github/docs/issues/2617" class="inline-onebox">Document all the GHCR labels which can be used · Issue #2617 · github/docs · GitHub</a></p>

**Comment by robinjhector** [comment]
<p>Hey! This is great :slight_smile:</p>
<p>Is there a way to make this work for Maven packages as well?<br>
My idea is to make automated releases of my artifact via Github Actions. I would like the newly published package to contain the changelog for the release.</p>

**Comment by clarkbw** [comment]
<p>No support for Maven yet but eventually our Maven service will be updated to use the new platform the Container registry is using now. At that time we should be able to better support operations like what you’re asking for.</p>
